### PR TITLE
2.9 into develop

### PR DESCRIPTION
**Pull request by ycliuhw**
```
Conflicts:
    agent/agentbootstrap/bootstrap_test.go
    apiserver/facades/client/charms/services/mocks/interface_mocks.go
    apiserver/facades/client/charms/services/repofactory_test.go
    apiserver/facades/controller/charmdownloader/mocks/mocks.go
    cmd/juju/application/deployer/bundlehandler_test.go
    cmd/juju/commands/bootstrap_test.go
    cmd/juju/commands/synctools.go
    cmd/juju/commands/synctools_test.go
    cmd/juju/commands/upgrademodel.go
    cmd/juju/commands/upgrademodel_test.go
    cmd/juju/commands/upgradecontroller_test.go
    cmd/jujud/agent/agenttest/agent.go
    cmd/jujud/agent/bootstrap_test.go
    cmd/jujud/agent/mongo_test.go
    core/charm/downloader/mocks/downloader_mocks.go
    environs/tools/build.go
    featuretests/bakerystorage_test.go
    featuretests/dblog_test.go
    featuretests/initiate_replset_test.go
    go.mod
    go.sum
    juju/api_test.go
    juju/testing/conn.go
    mongo/admin_test.go
    mongo/oplog_test.go
    provider/azure/environ_test.go
    provider/dummy/environs.go
    provider/dummy/environs_test.go
    state/backups/restore_test.go
    state/binarystorage/binarystorage_test.go
    state/cloudimagemetadata/image_test.go
    state/controller_test.go
    state/globalclock/reader_test.go
    state/globalclock/updater_test.go
    state/imagestorage/image_test.go
    state/initialize_test.go
    state/internal_test.go
    state/logs_internal_test.go
    state/model_test.go
    state/raftlease/target_test.go
    state/state_test.go
    state/storage/storage_test.go
    state/testing/conn.go
    state/testing/suite.go
    state/testing/suite_wallclock.go
    state/watcher/txnwatcher_test.go
    testing/cert.go
    testing/mgo.go
    worker/apiserver/worker_state_test.go
    worker/controllerport/manifold_test.go
    worker/globalclockupdater/manifold_test.go
    worker/httpserver/manifold_test.go
    worker/httpserver/state_test.go
    worker/httpserver/worker_test.go
    worker/raft/raftforwarder/manifold_test.go
    worker/state/manifold_test.go
    worker/upgrader/upgrader.go

```

- https://github.com/juju/juju/pull/14384
- https://github.com/juju/juju/pull/14365
- https://github.com/juju/juju/pull/14381
- https://github.com/juju/juju/pull/14344
- https://github.com/juju/juju/pull/14386
- https://github.com/juju/juju/pull/14392
- https://github.com/juju/juju/pull/14362
- https://github.com/juju/juju/pull/14393

### CODE REVIEW

File: rpc/params/model.go [hunk 1]
```diff
@@ -489,10 +489,16 @@ type ModelParam struct {
 }
 
 // UpgradeModel contains the arguments for UpgradeModel API call.
-type UpgradeModel struct {
+type UpgradeModelParams struct {
 	ModelTag            string         `json:"model-tag"`
-	ToVersion           version.Number `json:"to-version"`
+	TargetVersion       version.Number `json:"targert-version"`
```
Comment (hpidcock) [inline rpc/params/model.go:494]: targert?

**Review by hpidcock — APPROVED** [review]
just a typo in UpgradeModelParams that needs fixing

**Review by wallyworld — APPROVED** [review]

File: rpc/params/model.go (hunk 1 above)
Comment (ycliuhw) [reply rpc/params/model.go:494]: ops, thanks for fixing this!

**Pull request merged**